File: examples/pb-net-grpc/Server_FS/MyCalculator.fs
namespace Server_FS

open Shared_FS
open System.Threading.Tasks

type MyCalculator() =
    interface ICalculator with 
        member this.MultiplyAsync(request : MultiplyRequest) =
            let result = new MultiplyResult( Result = request.X * request.Y )
            new ValueTask<MultiplyResult>(result)

File: examples/pb-net-grpc/Server_VB/MyCalculator.vb
Imports Shared_VB
Public Class MyCalculator
    Implements ICalculator

    Function MultiplyAsync(request As MultiplyRequest) As ValueTask(Of MultiplyResult) Implements ICalculator.MultiplyAsync
        Dim result = New MultiplyResult With {.result = request.X * request.Y}
        Return New ValueTask(Of MultiplyResult)(result)
    End Function

End Class
